front/lib/api/resource_wrappers.ts

```diff
@@ -4,47 +4,57 @@ import type { NextApiRequest, NextApiResponse } from "next";
 import { Authenticator } from "@app/lib/auth";
 import type { SessionWithUser } from "@app/lib/iam/provider";
 import { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import { DataSourceViewResource } from "@app/lib/resources/data_source_view_resource";
 import { SpaceResource } from "@app/lib/resources/space_resource";
 import { apiError } from "@app/logger/withlogging";
 
 // This is a type that represents the resources that can be extracted from an API route
 type KeyToResource = {
   space: SpaceResource;
   dataSource: DataSourceResource;
+  dataSourceView: DataSourceViewResource;
 };
 
-type ResourceMap = { [K in ResourceKey]: KeyToResource[K] };
+type ResourceMap<U extends ResourceKey> = {
+  [K in U]: KeyToResource[K];
+};
+
+type OptionsMap<U extends ResourceKey> = {
+  [K in U]:
+    | {
+        requireCanRead?: boolean;
+        requireCanList?: boolean;
```

Comment:
Should we add requireCanWrite and requireCanAdmin right away to make it future proof and ready for other use-cases?